gsrs-fda-substance-extension/src/main/java/fda/gsrs/substance/exporters/ClinicalTrialEuropeDTOExporter.java

```diff
@@ -0,0 +1,221 @@
+
+        DEFAULT_RECIPE_MAP.put(ClinicalTrialEuropeDefaultColumns.SUBSTANCE_NAME, SingleColumnValueRecipe.create( ClinicalTrialEuropeDefaultColumns.SUBSTANCE_NAME ,(s, cell) ->{
+            StringBuilder sb = getClinicalTrialEuropeDrugDetails(s, ClinicalTrialEuropeDefaultColumns.SUBSTANCE_NAME);
+            cell.writeString(sb.toString());
+        }));
+        DEFAULT_RECIPE_MAP.put(ClinicalTrialEuropeDefaultColumns.SUBSTANCE_KEY, SingleColumnValueRecipe.create( ClinicalTrialEuropeDefaultColumns.SUBSTANCE_KEY ,(s, cell) ->{
+            StringBuilder sb = getClinicalTrialEuropeDrugDetails(s, ClinicalTrialEuropeDefaultColumns.SUBSTANCE_KEY);
+            cell.writeString(sb.toString());
+        }));
+
+        DEFAULT_RECIPE_MAP.put(ClinicalTrialEuropeDefaultColumns.CONDITIONS, SingleColumnValueRecipe.create( ClinicalTrialEuropeDefaultColumns.CONDITIONS ,(s, cell) ->{
+            StringBuilder sb = getMeddraTermDetails(s, ClinicalTrialEuropeDefaultColumns.CONDITIONS);
+            cell.writeString(sb.toString());
+        }));
+
+        DEFAULT_RECIPE_MAP.put(ClinicalTrialEuropeDefaultColumns.SPONSOR_NAME, SingleColumnValueRecipe.create( ClinicalTrialEuropeDefaultColumns.SPONSOR_NAME ,(s, cell) ->{
+            cell.writeString(s.getSponsorName());
+        }));
+
+        DEFAULT_RECIPE_MAP.put(ClinicalTrialEuropeDefaultColumns.RESULTS, SingleColumnValueRecipe.create( ClinicalTrialEuropeDefaultColumns.RESULTS,(s, cell) ->{
+            cell.writeString(s.getTrialResults());
+        }));
+    }
+
+    private static StringBuilder getClinicalTrialEuropeDrugDetails(ClinicalTrialEuropeDTO s, ClinicalTrialEuropeDefaultColumns fieldName) {
+        StringBuilder sb = new StringBuilder();
+        try {
+            List<ClinicalTrialEuropeProductDTO> pList= s.getClinicalTrialEuropeProductList();
+            if (pList!=null && !pList.isEmpty()) {
+                for (ClinicalTrialEuropeProductDTO p : pList) {
+                    List<ClinicalTrialEuropeDrugDTO> dList = p.getClinicalTrialEuropeDrugList();
+                    for (ClinicalTrialEuropeDrugDTO d : dList) {
+                        if (sb.length() != 0) {
+                            sb.append("|");
+                        }
+                        switch (fieldName) {
+                            case SUBSTANCE_KEY:
+                                sb.append((d.getSubstanceKey() != null) ? d.getSubstanceKey() : "(No Substance Key)");
+                                break;
+                            case SUBSTANCE_NAME:
+                                Optional<Substance> substanceEntity = substanceEntityService.get(UUID.fromString(d.getSubstanceKey()));
+                                String ptUTF8 = substanceEntity.get().getDisplayName()
+                                        .map(n -> n.getName())
+                                        .orElse("(No Substance Name)");
+                                sb.append(ptUTF8);
+                                break;
```

Comment:
Should use a cache. Ideally gsrsCache. 

I'd add this to the top of this class:

```
private static GsrsCache gsrsCache;
```

And then change this section to be:

```
if(gsrsCache  ==null){
   gsrsCache  = StaticContextAccessor.getBean(GsrsCache.class);
}
String nm = gsrsCache.getOrElse("SKEY:" + d.getSubstanceKey(), ()->{
 Optional<Substance> substanceEntity = substanceEntityService.get(UUID.fromString(d.getSubstanceKey()));
return substanceEntity.get().getDisplayName()
                                        .map(n -> n.getName())
                                        .orElse("(No Substance Name)");
                                        
});
sb.append(nm);
```